rfcs/0000-rfc-process.md

```diff
@@ -0,0 +1,135 @@
+ecosystem is evolving in.
+
+# Motivation
+[motivation]: #motivation
+
+There are a number of changes that are significant enough that they could
+benefit from wider community consensus before being implemented. Either
+because they introduce new concepts, big changes or are controversial enough
+that not everybody will agree on the direction to take.
+
+# Detailed design
+[design]: #detailed-design
+
+Many changes, including bug fixes and documentation improvements can be
+implemented and reviewed via the normal GitHub pull request workflow.
+
+Some changes though are "substantial", and we ask that these be put through a
+bit of a design process and produce a consensus among the Nix community.
+
+This is the bulk of the RFC. Explain the design in enough detail for somebody
+familiar with the ecosystem to understand, and implement.  This should get
+into specifics and corner-cases, and include examples of how the feature is
+used.
+
+## When you need to follow this process
+
+You need to follow this process if you intend to make "substantial" changes to
+the Nix ecosystem. What constitutes a "substantial" change is evolving based
+on community norms, but may include the following.
+
+  - Any semantic or syntactic change to the language that is not a bugfix.
+  - Removing language features
+  - Big restructuring of nixpkgs
+  - Expansions to the scope of nixpkgs (new arch, major subprojects, ...)
+  - Introduction of new interfaces or functions
+  - A controversial change
+
+Some changes do not require an RFC:
+
+  - Adding, updating and removing packages in nixpkgs
+  - Fixing security updates and bugs that don't break interfaces
+
+If you submit a pull request to implement a new feature without going
+through the RFC process, it may be closed with a polite request to
+submit an RFC first.
+
+## What the process is
+
+In short, to get a major feature added to the Nix ecosystem, one should first
+go throught the RFC process in order to improve the likelyhood of inclusion.
+Here are roughly the steps that one would take:
+
+* Fork the RFC repo https://github.com/NixOS/rfcs
+* Copy `0000-template.md` to `rfcs/0000-my-feature.md` (where
+'my-feature' is descriptive. don't assign an RFC number yet).
+* Fill in the RFC
+* Submit a pull request. The pull request is the time to get review of
+the design from the larger community.
+* Build consensus and integrate feedback. RFCs that have broad support
+are much more likely to make progress than those that don't receive any
+comments.
+
+At this point, the person submitting the RFC should find at least one "buddy"
+that will help them bring the RFC to reality. The goal is to improve the
+chances that the RFC is both desired and likely to be implemented.
+
+Whomever merges the RFC should do the following:
+
+* Assign an id, using the PR number of the RFC pull request. (If the RFC
+  has multiple pull requests associated with it, choose one PR number,
+  preferably the minimal one.)
```

Comment:
Ok that's the reason why the RFC needs to be either renamed at the very start or very end: otherwise all the existing comments are being hidden by github. This will be part of the CONTRIBUTING.md or PULL_REQUEST.md so users do the right thing.